src/components/charts/SwipeChart.tsx

```diff
@@ -0,0 +1,554 @@
+  }, [sPulseRMultiplier])
+
+  const aInnerPulseStyle: Animated.AnimateProps<CircleProps> = useAnimatedProps(() => ({
+    r: sPulseRMultiplier.value * 6,
+    opacity: sCursorOpacity.value
+  }))
+  const aOuterPulseStyle: Animated.AnimateProps<CircleProps> = useAnimatedProps(() => ({
+    r: 6 + sPulseRMultiplier.value * 6,
+    opacity: (1 - sPulseRMultiplier.value) * sCursorOpacity.value
+  }))
+
+  // X axis labels
+  const aXTooltipStyle = useAnimatedStyle(() => ({
+    left: sXTooltipPos.value,
+    opacity: sCursorOpacity.value
+  }))
+  const aXEndLabelStyle = useAnimatedStyle(() => ({
+    opacity: 1 - sCursorOpacity.value
+  }))
+
+  // #endregion Animations
+
+  // #region Handlers
+
+  const handleGradient = useHandler((props: GradientProps) => {
+    return (
+      <LinearGradient x1="50%" y1="0%" x2="50%" y2="100%" {...props}>
+        <Stop stopColor={theme.iconTappable} offset="0%" stopOpacity="0.5" />
+        <Stop stopColor={theme.iconTappable} offset="100%" stopOpacity="0" />
+      </LinearGradient>
+    )
+  })
+
+  // sIsShowCursor is used as an optimization since handleToolTipTextRenderer()
+  // is triggered repeatedly during the price line animation when chart dataset
+  // is updated during scene initialization or timespan changes. This can cause
+  // sluggishness during those updates.
+  // We only care about those callbacks when it's triggered by the gesture, not
+  // by the builtin price line entering animation sequence.
+  const handleShowIndicatorCallback = useHandler((opacity: number) => {
+    const isShowIndicator = opacity === 1
+    rIsShowCursor.current = isShowIndicator
+    sCursorOpacity.value = withTiming(isShowIndicator ? 1 : 0, { duration: isShowIndicator ? 300 : 500 })
+  })
+
+  // Update a shared X position equal to that of the active slide gesture
+  const handleToolTipTextRenderer = useHandler((toolTipTextRenderersInput: ToolTipTextRenderersInput) => {
+    if (rIsShowCursor.current) sXTooltipPos.value = toolTipTextRenderersInput.x
+
+    // The SDK's API requires this return value even though we're not using it.
+    return { text: '' }
+  })
+
+  // X axis date tooltip display value updates
+  const handleDateCallbackWithX = useHandler((x: number | Date) => {
+    if (rIsShowCursor.current) {
+      const newXTooltipText = formatTimestamp(new Date(x), selectedTimespan).xTooltip
+      if (sXTooltipString.value !== newXTooltipText) sXTooltipString.value = newXTooltipText
+    }
+  })
+
+  // Price value tooltip display value updates
+  const handlePriceCallbackWithY = useHandler((y: number) => {
+    if (rIsShowCursor.current) {
+      const newDisplayPrice = formatFiatString({ fiatAmount: y.toString(), noGrouping: true, autoPrecision: true })
+      if (newDisplayPrice !== sPriceValString.value) sPriceValString.value = newDisplayPrice
+    }
+  })
+
+  // Natively align a component based on a specified origin
+  const handleAlignLayout = (origin: AlignOrigin, offset?: number) => {
```

Comment:
You use `handleAlignLayout` in 4 places but it creates a new function declaration each time. Instead create a memoized function with useCallback for each use case.